ctapipe/image/modifications.py

```diff
@@ -0,0 +1,113 @@
+from abc import abstractmethod
+import numpy as np
+from ..core.component import TelescopeComponent
+from ..core.traits import FloatTelescopeParameter, BoolTelescopeParameter
+
+
+def add_noise(image, noise_level, rng=None, correct_bias=True):
+    """
+    Create a new image with added poissonian noise
+    """
+    if not rng:
+        rng = np.random.default_rng()
+    noisy_image = image.copy()
+    noise = rng.poisson(noise_level)
+    noisy_image += noise
+    if correct_bias:
+        noisy_image -= noise_level
+    return noisy_image
+
+
+def smear_image(image, geom, smear_factor):
+    """
+    Create a new image with values smeared to the direct pixel neighbors
+    Pixels at the camera edge lose charge this way.
+    """
+    # make clear what smear factor is supposed to mean and that only direct neighbors are taken into account
+    # smear factor can be an array as well -> selection in tool possible!
+    # a more sophisticated approach might make use of the pixel area, but thats complicated
+    if geom.pix_type.value == "hexagon":
+        max_neighbors = 6
+    elif geom.pix_type.value == "rectangular":  # thats probably labeld differently
+        max_neighbors = 8  # or 4? lookup neighbor matrix for a rect cam
+    else:
+        print(geom.pix_type)
+        raise Exception("Unknown pixel type")
+
+    diffused_image = (
+        (image * geom.neighbor_matrix).sum(axis=1) * smear_factor / max_neighbors
+    )
+    remaining_image = image * (1 - smear_factor)
+    smeared_image = remaining_image + diffused_image
+    return smeared_image
+
+
+class ImageModifier(TelescopeComponent):
+    """
+    Abstract class for configurable image modifying algorithms. Use
+    ``ImageModifier.from_name()`` to construct an instance of a particular algorithm
+    """
+
+    @abstractmethod
+    def __call__(self, tel_id: int, image: np.ndarray) -> np.ndarray:
+        """
+        Modifies a given image
+
+        Returns
+        -------
+        np.ndarray
+            modified image
+        """
+        pass
+
+
+class NullModifier(ImageModifier):
+    def __call__(self, tel_id: int, image: np.ndarray) -> np.ndarray:
+        return image
+
+
+class LSTImageModifier(ImageModifier):
```

Comment:
This is not really specific to LST I think.

Reply:
Not really, its just the current lstchain procedure.
I was not sure if we wanted to mirror this behaviour 1to1, so I marked in in the name for now.